tests/test_applications.py

```diff
@@ -534,3 +534,14 @@ def get_app() -> ASGIApp:
     test_client_factory(app).get("/foo")
 
     assert SimpleInitializableMiddleware.counter == 2
+
+
+def test_lifespan_typing():
+    class App(Starlette):
+        pass
+
+    @asynccontextmanager
+    async def lifespan(app: App) -> AsyncIterator[None]:
+        yield
+
+    App(lifespan=lifespan)
```

Comment:
Not sure if we actually enforce mypy on our tests, but pylance reports no issues